client/src/actions/team.js

```diff
@@ -12,44 +13,47 @@ import {
   DELETE_TEAM_MEMBER,
   ADD_TEAM,
 } from './types';
+import { errorActionGlobalCreator } from '../error/errorReduxFunctions';
 
 /**
  * Adds a new team to redux store and database.
  * @param teamInfo contains teamName, orgName and email
  */
 export const addTeamInfo = (teamInfo) => async (dispatch) => {
-  const data = await api.addTeam(teamInfo);
-  const teamData = {
-    ...teamInfo,
-    teamId: data.data._id,
-  };
-  dispatch({
-    type: ADD_TEAM,
-    payload: teamData,
-  });
+  try {
+    const data = await api.addTeam(teamInfo);
+    const teamData = {
+      ...teamInfo,
+      teamId: data.data._id,
+    };
+    dispatch({
+      type: ADD_TEAM,
+      payload: teamData,
+    });
+  } catch (err) {
+    dispatch(errorActionGlobalCreator(err));
+  }
 };
 
 /**
  * Finds a team
  * @param teamCredentials team email and password as a dictionary
  * @param teamPassword team account password
  */
-export const getTeam = (teamCredentials) => async (dispatch) => {
-  const data = await api.loginTeam(teamCredentials);
-  const teamData = data.data.team;
-  console.log(teamData);
-  const team = {
-    teamId: teamData._id,
-    email: teamData.email,
-    teamName: teamData.teamName,
-    orgName: teamData.orgName,
-    twitterHandle: teamData.twitterHandle,
-    repoCreated: teamData.repoCreated,
-  };
-  dispatch({
-    type: ADD_TEAM,
-    payload: team,
-  });
+export const getTeam = (teamCredentials, history) => async (dispatch) => {
+  try {
+    const data = await api.loginTeam(teamCredentials);
+    console.log(data);
+    const teamData = data.data.team;
+    const team = teamDataAllocator(teamData);
+    dispatch({
+      type: ADD_TEAM,
+      payload: team,
+    });
+    history.push('/dashboard');
```

Comment:
Would changing page from here be consistent with React guidelines? I assumed it's something the React component should do

Reply:
Thanks for pointing it out! 💯  I totally forgot to revert it back to the original from my little experiment. I update it in my new commit below. 😆